src/group_theory/divisible.lean

```diff
@@ -253,3 +262,22 @@ noncomputable instance quotient_group.rootable_by [rootable_by A ℕ] : rootable
 quotient_group.mk_surjective.rootable_by _ $ λ _ _, rfl
 
 end quotient
+
+section rat_circle
+
+universe u
+
+/--
+The rational circle in universe u defined as `ℚ ⧸ ℤ`
+-/
+def rat_circle : Type u := (ulift.{u} (ℚ ⧸ (algebra_map ℤ ℚ).to_add_monoid_hom.range))
```

Comment:
You may use [int.cast_add_hom](https://leanprover-community.github.io/mathlib_docs/data/int/cast.html#int.cast_add_hom)